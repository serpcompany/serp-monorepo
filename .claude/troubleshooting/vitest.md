https://vitest.dev/guide/common-errors.html