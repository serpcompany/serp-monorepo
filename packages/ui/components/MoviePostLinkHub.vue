<script setup lang="ts">
const data = await usePosts(1, 50, '', 'movies', true)
</script>

<template>
  <UPageList divide>
    <UPageCard
      v-for="post in data?.posts"
      :key="post.slug"
      variant="ghost"
      :to="`/movies/${post.slug}`"
    >
      <template #body>
        <UUser
          :name="post.keyword"
          :avatar="{ src: post.featuredImage }"
          size="xl"
        />
      </template>
    </UPageCard>
  </UPageList>
</template>
