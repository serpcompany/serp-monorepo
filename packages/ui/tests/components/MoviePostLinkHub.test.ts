/* eslint-disable @typescript-eslint/no-unused-vars, no-unused-vars  */

import { describe, expect, it } from 'vitest';
import MoviePostLinkHub from '../../components/MoviePostLinkHub.vue';
import ComponentRender from '../componentRender';

describe('MoviePostLinkHub Snapshot', () => {
  const scenarios: [
    string,
    { posts: Array<{ slug: string; keyword: string; featuredImage: string }> }
  ][] = [
    [
      'with posts',
      {
        posts: [
          {
            slug: 'movie-1',
            keyword: 'Movie One',
            featuredImage: 'https://example.com/movie1.jpg'
          },
          {
            slug: 'movie-2',
            keyword: 'Movie Two',
            featuredImage: 'https://example.com/movie2.jpg'
          }
        ]
      }
    ],
    ['with no posts', { posts: [] }]
  ];

  it.each(scenarios)('renders %s correctly', async (desc, { posts }) => {
    (globalThis as unknown).usePosts = async (...args: unknown) => ({ posts });

    const html = await ComponentRender(
      `MoviePostLinkHub ${desc}`,
      {},
      MoviePostLinkHub
    );
    expect(html).toMatchSnapshot();
  });
});
