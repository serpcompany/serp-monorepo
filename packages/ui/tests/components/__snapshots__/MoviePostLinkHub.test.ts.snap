// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`moviePostLinkHub Snapshot > renders with no posts correctly 1`] = `
<div
  class="
    *:not-last:after:absolute
    *:not-last:after:bg-border
    *:not-last:after:bottom-0
    *:not-last:after:h-px
    *:not-last:after:inset-x-1
    flex
    flex-col
    relative
  "
  role="list"
></div>
`;

exports[`moviePostLinkHub Snapshot > renders with posts correctly 1`] = `
<div
  class="
    *:not-last:after:absolute
    *:not-last:after:bg-border
    *:not-last:after:bottom-0
    *:not-last:after:h-px
    *:not-last:after:inset-x-1
    flex
    flex-col
    relative
  "
  role="list"
>
  <div
    class="
      flex
      hover:bg-elevated/50
      relative
      rounded-lg
      transition
    "
    data-orientation="vertical"
  >
    <!-- v-if -->
    <div class="
      flex
      flex-1
      flex-col
      gap-x-8
      gap-y-4
      lg:grid
      p-4
      relative
      sm:p-6
    ">
      <div class="
        flex
        flex-1
        flex-col
        items-start
      ">
        <!-- v-if -->
        <!-- v-if -->
        <div class="flex-1">
          <div
            class="
              flex
              gap-2.5
              group/user
              items-center
              relative
            "
            data-orientation="horizontal"
          >
            <span class="
              align-middle
              bg-elevated
              inline-flex
              items-center
              justify-center
              overflow-hidden
              rounded-full
              select-none
              shrink-0
              size-10
              text-xl
            ">
              <img
                alt="Movie One"
                class="
                  h-full
                  object-cover
                  rounded-[inherit]
                  w-full
                "
                data-nuxt-img=""
                height="40"
                role="img"
                src="https://example.com/movie1.jpg"
                srcset="https://example.com/movie1.jpg 1x, https://example.com/movie1.jpg 2x"
                width="40"
              />
            </span>
            <div class="">
              <!-- v-if -->
              <p class="
                font-medium
                text-base
                text-highlighted
              ">
                Movie One
              </p>
              <!-- v-if -->
            </div>
          </div>
        </div>
        <!-- v-if -->
      </div>
    </div>
    <a
      aria-label="Card link"
      class="
        focus:outline-none
        peer
      "
      href="/movies/movie-1"
      tabindex="-1"
    ><span
      aria-hidden="true"
      class="
        absolute
        inset-0
      "
    ></span></a>
  </div>
  <div
    class="
      flex
      hover:bg-elevated/50
      relative
      rounded-lg
      transition
    "
    data-orientation="vertical"
  >
    <!-- v-if -->
    <div class="
      flex
      flex-1
      flex-col
      gap-x-8
      gap-y-4
      lg:grid
      p-4
      relative
      sm:p-6
    ">
      <div class="
        flex
        flex-1
        flex-col
        items-start
      ">
        <!-- v-if -->
        <!-- v-if -->
        <div class="flex-1">
          <div
            class="
              flex
              gap-2.5
              group/user
              items-center
              relative
            "
            data-orientation="horizontal"
          >
            <span class="
              align-middle
              bg-elevated
              inline-flex
              items-center
              justify-center
              overflow-hidden
              rounded-full
              select-none
              shrink-0
              size-10
              text-xl
            ">
              <img
                alt="Movie Two"
                class="
                  h-full
                  object-cover
                  rounded-[inherit]
                  w-full
                "
                data-nuxt-img=""
                height="40"
                role="img"
                src="https://example.com/movie2.jpg"
                srcset="https://example.com/movie2.jpg 1x, https://example.com/movie2.jpg 2x"
                width="40"
              />
            </span>
            <div class="">
              <!-- v-if -->
              <p class="
                font-medium
                text-base
                text-highlighted
              ">
                Movie Two
              </p>
              <!-- v-if -->
            </div>
          </div>
        </div>
        <!-- v-if -->
      </div>
    </div>
    <a
      aria-label="Card link"
      class="
        focus:outline-none
        peer
      "
      href="/movies/movie-2"
      tabindex="-1"
    ><span
      aria-hidden="true"
      class="
        absolute
        inset-0
      "
    ></span></a>
  </div>
</div>
`;
